front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -36,7 +36,7 @@ const useScenarioData = (scenario: ScenarioWithDetails, infraId: number) => {
 
   const [timetableItems, setTimetableItems] = useState<TimetableItem[]>();
   const timetableItemsById = useMemo(() => mapBy(timetableItems, 'id'), [timetableItems]);
-  const [selectedTimetableItemIds, setSelectedTimetableItemIds] = useState<TimetableItemId[]>([]);
+  const [selectedTimetableItems, setSelectedTimetableItems] = useState<TimetableItem[]>([]);
```

Comment:
I don't think storing the full selected timetable item is a good idea here. The stored data will be outdated as soon as the original item is mutated. If a paced train is selected, then turned into a one-time train by editing it (via NGE or the OSRD modal), `selectedTimetableItems` will not be updated. (This is true for other train properties as well.)

Can we instead keep only the ID in `selectedTimetableItems`, and grab the full timetable item from `timetableItemsById` when we need it?

Reply:
Ok, I can drop this commit :+1: